Docstring fix & small adjustements in `demo_affine_image`

afvincent opened the pull request:
1/ Makes the red rectangle announced in the docstring red in the plot. The docstring has also been precised a bit.

2/ I think the comment about the image rotation was misplaced so I moved it where it seemed relevant to me.

3/ Finally, I've made the line width of the rectangle a bit bigger to be more visible and added a call to `plt.show()`

The example should now looks like:
![modified_demo_affine_image](https://cloud.githubusercontent.com/assets/17270724/16533777/6632f1ae-3fdb-11e6-9b28-9ce547b12eef.png)

**Remark 1**: the colormap is not `viridis` because the colormap seems to default to `gnuplot2` in my conda env. I guess I had tweaked sthg and didn't remember it.

**Remark 2**: the red rectangle does not seem to be perfectly centered on the image: the 2 upper sides are mostly over the image, while the 2 lower sides are mostly out of it. Is it expected?

QuLogic commented on examples/api/demo_affine_image.py:
```diff
@@ -23,23 +23,24 @@ def get_image():
 
     # display intended extent of the image
     x1, x2, y1, y2 = im1.get_extent()
     x3, y3 = x2, y1
```
Are these used for anything?

tacaswell commented:
The offset between the image and rectangle does seem a bit odd...

afvincent commented:
@QuLogic  You're right, and I didn't pay attention enough: x3 and y3 weren't used anywhere. I also removed the import of `matplotlib.cm` that was also not used.

I had an idea during the night and significantly modified the example to also show the image extent in the case where the affine transform is not applied. I've factorize the plotting of the extent rectangle and added some segment markers. However, the latters might be a bit overkill if one doesn't add a translation to the affine transorm (currently it's only a rotation).

The commit 2f263b5 produces:
![modified_demo_affine_image_v1](https://cloud.githubusercontent.com/assets/17270724/16539434/9b71f6c2-4043-11e6-90f3-9e7f2583c532.png)

@tacaswell I'll try to investigate the possible issue of the offset between the image and the rectangle (I guess I'll have to learn about the details of Affine2D first…).

afvincent commented:
PR #6792 is not only about `demo_affine_image` but as it has replicated the docstring fixes and will provide an extended demo of the affine transformations that may be more relevant than my changes, I think it should supersede the current PR. So I'm going to close this one. Please reopen it if anyone thinks it was a mistake.
